DataMigration/src/test/java/org/sagebionetworks/dian/datamigration/BridgeJavaSdkUtilTests.java

```diff
@@ -0,0 +1,455 @@
+import com.google.common.collect.ImmutableMap;
+
+import org.apache.commons.lang3.StringUtils;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+import org.sagebionetworks.bridge.rest.api.ForResearchersApi;
+import org.sagebionetworks.bridge.rest.api.ParticipantReportsApi;
+import org.sagebionetworks.bridge.rest.api.ParticipantsApi;
+import org.sagebionetworks.bridge.rest.exceptions.EntityNotFoundException;
+import org.sagebionetworks.bridge.rest.model.IdentifierHolder;
+import org.sagebionetworks.bridge.rest.model.Message;
+import org.sagebionetworks.bridge.rest.model.SharingScope;
+import org.sagebionetworks.bridge.rest.model.SignUp;
+import org.sagebionetworks.bridge.rest.model.StudyParticipant;
+
+import java.io.IOException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.HashMap;
+import java.util.Map;
+
+import retrofit2.Call;
+import retrofit2.Response;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertTrue;
+import static org.sagebionetworks.dian.datamigration.BridgeJavaSdkUtil.*;
+
+public class BridgeJavaSdkUtilTests extends Mockito {
+
+    private final Path resourceDirectory = Paths.get("src", "test", "resources");
+    private final Path testSessionSchedulePath = resourceDirectory
+            .resolve("testSessionSchedules")
+            .resolve("test_session_schedules_2021-07-08")
+            .resolve("000000 test_session_schedule 2019-08-29T16-06-11Z.json");
+    private final Path wakeSleepSchedulePath = resourceDirectory
+            .resolve("wakeSleepSchedules")
+            .resolve("wake_sleep_schedules_08-07-21")
+            .resolve("000000 Availability 2019-08-29T16-06-11Z.json");
+
+    @Mock
+    private ForResearchersApi mockResearcherApi;
+
+    @Mock
+    private ParticipantReportsApi mockReportsApi;
+
+    @Mock
+    private ParticipantsApi mockParticipantsApi;
+
+    @Mock
+    private Call<Message> mockTestSessionReportCall;
+
+    @Mock
+    private Call<Message> mockWakeSleepReportCall;
+
+    @Mock
+    private Call<Message> mockCompletedTestsReportCall;
+
+    @Mock
+    private Call<Message> mockDeleteTestSessionReportCall;
+
+    @Mock
+    private Call<Message> mockDeleteWakeSleepReportCall;
+
+    @Mock
+    private Call<Message> mockDeleteCompletedTestReportCall;
+
+    @Mock
+    private Call<Message> mockUpdateParticipantCall;
+
+    @Mock
+    private Call<IdentifierHolder> mockSignUpCall;
+
+    @Mock
+    private Call<StudyParticipant> mockGetExternalIdMigratedCall;
+
+    @Mock
+    private Call<StudyParticipant> mockGetExternalIdNotMigratedCall;
+
+    @Before
+    public void before() throws IOException {
+        if (mockResearcherApi == null) {  // Only initialize once
```

Comment:
I avoided `@BeforeClass` because it was _static_.  I'm new-ish to Mockito, so I thought it was bad practice to setup before each test, but I removed the null conditional and it works the same and does not same any less efficient.

Reply:
Setting up before each test is good.  It makes sure each test starts from a known state.